lib/src/hooks/types.ts

```diff
@@ -24,12 +24,101 @@ export type UseHalResource = {
 
 export type RequestStatus = "idle" | "fetching" | "resolved" | "rejected" | "interaction";
 
+type InteractionType = Record<string, unknown> & {
```

Comment:
Hi @marcialfps , We believe it would be better to move everything from line 27 to 109 to a separate Halstack -client types library. 

Also will investigate how to contribute to the public npm or @types and push all the types we are exporting from Halstack libraries, including hasltack-react-hal and even the halstack-react-cdk. 

We will start and spike to learn more about @types and agree on the best approach moving forward. I would keep this PR on Hold for the time being if that's ok with you.